### PR TITLE
validate ocs deployment with vault

### PR DESCRIPTION
**Pull request by shylesh**
Post deployment verification for OCS with vault

Signed-off-by: Shylesh Kumar Mohan <shmohan@redhat.com>

### CODE REVIEW

**Comment by shylesh** [comment]
Rebased on latest master

File: ocs_ci/ocs/resources/storage_cluster.py [hunk 1]
```diff
@@ -378,6 +378,9 @@ def ocs_install_verification(
         check_fips_enabled()
     if config.ENV_DATA.get("encryption_at_rest"):
         osd_encryption_verification()
+        if config.DEPLOYMENT.get("kms_deployment"):
+            if config.ENV_DATA.get("KMS_PROVIDER") == constants.VAULT_KMS_PROVIDER:
```
Comment (clacroix12) [inline ocs_ci/ocs/resources/storage_cluster.py:382]: Should we have a generic `post_deploy_verification` method in the `KMS` class that calls the provider-specific method based on `config.ENV_DATA.get("KMS_PROVIDER")`? This way we don't need to check for each provider type in this code (code that really doesn't care about WHICH kms provider we are using).

Comment (shylesh) [reply ocs_ci/ocs/resources/storage_cluster.py:382]: I guess you meant like this. Makes sense, I will do it.
```python
if config.DEPLOYMENT.get("kms_deployment"):
    kms = KMS.get_kms_deployment()
    kms.post_deploy_verification()
```

**Review by clacroix12 — APPROVED** [review]

**Review by vavuthu — DISMISSED** [review]

**Comment by shylesh** [comment]
Verified local with and without Namespaces
=================================

Without namespaces
----------------------------
13:10:07 - MainThread - ocs_ci.utility.kms - INFO - Created cert file at /mnt/shmohan/Downloads/git/ocs-ci/data/vault-client.crt
13:10:07 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get ConfigMap ocs-kms-connection-details -n openshift-storage -o yaml
13:10:07 - MainThread - ocs_ci.utility.kms - INFO - setting vault_backend_path = ocs-fef19929-f1d6-4469-a185-8134c1e3785d
13:10:08 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get Pod  -n openshift-storage --selector=app=rook-ceph-osd -o yaml
13:10:10 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get Pod rook-ceph-osd-0-57579fc9cb-pcjl5 -n openshift-storage -o yaml
13:10:10 - MainThread - ocs_ci.utility.kms - INFO - Vault: Found key for ocs-deviceset-2-data-0sdct6
13:10:10 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get Pod rook-ceph-osd-1-69fdc4c596-4z96w -n openshift-storage -o yaml
13:10:11 - MainThread - ocs_ci.utility.kms - INFO - Vault: Found key for ocs-deviceset-0-data-0wctjc
13:10:11 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get Pod rook-ceph-osd-2-cfc74c98d-dn59j -n openshift-storage -o yaml
13:10:12 - MainThread - ocs_ci.utility.kms - INFO - Vault: Found key for ocs-deviceset-1-data-09lq4h
13:10:12 - MainThread - ocs_ci.utility.kms - INFO - Found Noobaa root secret path
13:10:12 - MainThread - ocs_ci.utility.kms - INFO - Checking if StorageCluster has configured KMS encryption



With Namespace
--------------------------
12:33:28 - MainThread - ocs_ci.utility.kms - INFO - Created cert file at /mnt/shmohan/Downloads/git/ocs-ci/data/vault-client.crt
12:33:28 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get ConfigMap ocs-kms-connection-details -n openshift-storage -o yaml
12:33:29 - MainThread - ocs_ci.utility.kms - INFO - Setting vault_namespace=ocs-namespace-c7c29a6b-8659-46e1-a413-ce7b62b1f7be
12:33:29 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get ConfigMap ocs-kms-connection-details -n openshift-storage -o yaml
12:33:29 - MainThread - ocs_ci.utility.kms - INFO - setting vault_backend_path = ocs-c7c29a6b-8659-46e1-a413-ce7b62b1f7be
12:33:30 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get Pod  -n openshift-storage --selector=app=rook-ceph-osd -o yaml
12:33:32 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get Pod rook-ceph-osd-0-867dd955bb-nrq7p -n openshift-storage -o yaml
12:33:33 - MainThread - ocs_ci.utility.kms - INFO - Vault: Found key for ocs-deviceset-2-data-08lxrb
12:33:33 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get Pod rook-ceph-osd-1-556487fcf5-w9mww -n openshift-storage -o yaml
12:33:34 - MainThread - ocs_ci.utility.kms - INFO - Vault: Found key for ocs-deviceset-0-data-0wkr4f
12:33:34 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get Pod rook-ceph-osd-2-8689b8cd59-qrqnf -n openshift-storage -o yaml
12:33:35 - MainThread - ocs_ci.utility.kms - INFO - Vault: Found key for ocs-deviceset-1-data-086sr2
12:33:35 - MainThread - ocs_ci.utility.kms - INFO - Found Noobaa root secret path
12:33:35 - MainThread - ocs_ci.utility.kms - INFO - Checking if StorageCluster has configured KMS encryption

**Comment by shylesh** [comment]
rebased on latest master

**Review by clacroix12 — APPROVED** [review]

**Review by petr-balogh — APPROVED** [review]

**Review by shreekarSS — APPROVED** [review]

**Review by rachael-george — APPROVED** [review]

**Pull request merged**